pitmutationmate-override-plugin/plugin/src/main/groovy/io/github/amosproj/pitmutationmate/override/strategy/DependencyInclusionStrategy.groovy

```diff
@@ -0,0 +1,80 @@
+// SPDX-FileCopyrightText: 2023 Lennart Heimbs
+//
+// SPDX-License-Identifier: MIT
+
+package io.github.amosproj.pitmutationmate.override.strategy
+
+import io.github.amosproj.pitmutationmate.override.PITSettingOverridePlugin
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+
+/**
+ * DependencyInclusionStrategy
+ *
+ * This class is responsible for adding a dependency to the pitest plugin extension.
+ * It is used by the [PITSettingOverridePlugin].
+ *
+ * @see [ PITSettingOverridePlugin ]
+ */
+class DependencyInclusionStrategy implements OverrideStrategy {
+
+    public static final String OVERRIDE_ATTRIBUTE = 'addCoverageListenerDependency'
+    static final String PITEST_EXTENSION = 'pitest'
+    static final String PITEST_PLUGIN = 'info.solidsoft.pitest'
+    static final String PITEST_PLUGIN_ANDROID = 'pl.droidsonroids.pitest'
+
+    @SuppressWarnings('FieldName')
+    private final static Logger log = Logging.getLogger(PITSettingOverridePlugin)
+
+    @Override
+    void apply(Project proj, String propertyName, String overrideValue) {
+        log.debug("Trying to add dependency '$overrideValue'.")
+
+        boolean isApplied = false
+        proj.gradle.allprojects { project ->
```

Comment:
Didn't the industry partner say we should use `proj.allprojects` instead of `proj.gradle.allprojects`? For me both works.

Reply:
Truthfully i forgot which one they said was better. I had a look at the docs and it sounded like the gradle one will include project that are added *later* as well so I chose that...

In my testing it didn't make a difference. In fact the `allprojects` part was not needed for the two sample projects either way but I'll keep it just in case and change it to `proj.allprojects`